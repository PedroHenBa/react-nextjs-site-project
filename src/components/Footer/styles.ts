import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-bottom: 0;
      padding-top: 0;
    }
  `}
`;
