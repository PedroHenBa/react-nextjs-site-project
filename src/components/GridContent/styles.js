import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    margin: 0 auto;
    max-width: 58rem;
  `}
`;

export const html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.big} 0;
  `}
`;
