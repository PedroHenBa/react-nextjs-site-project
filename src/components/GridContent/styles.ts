import styled, { css } from 'styled-components';

export const Container = styled.div`
  text-align: center;
  margin: 0 auto;
  max-width: 58rem;
`;

export const html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.big} 0;
  `}
`;
