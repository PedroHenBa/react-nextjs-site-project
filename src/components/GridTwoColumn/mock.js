export default {
  title: 'Grid two column',
  text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec quis enim magna.
    Nam sit amet ante vitae massa pellentesque volutpat. Vestibulum sagittis justo sit amet pretium inte
    rdum. Phasellus ornare.`,
  srcImage: 'assets/images/javascript.svg',
};
