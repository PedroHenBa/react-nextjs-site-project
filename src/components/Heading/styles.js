import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.big};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xbig};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xbig};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.whiteColor};
    ${titleSizes[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
