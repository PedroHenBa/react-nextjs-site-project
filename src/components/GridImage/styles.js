import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacing.xbig};
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.big};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const GridImage = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
