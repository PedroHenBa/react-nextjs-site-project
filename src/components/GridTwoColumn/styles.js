import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    gap: ${theme.spacing.big};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Heading} {
      margin-bottom: ${theme.spacing.xbig};
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.big};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
