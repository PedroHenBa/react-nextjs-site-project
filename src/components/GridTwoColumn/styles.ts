import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    gap: ${theme.spacing.big};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Heading} {
      margin-bottom: ${theme.spacing.xbig};
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.big};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
