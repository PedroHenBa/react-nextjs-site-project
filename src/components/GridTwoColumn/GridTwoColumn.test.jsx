import { GridTwoColumn } from './index';
import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import mock from './mock';

describe('<GridTwoColumn />', () => {
  it('should render Two Column Grid component', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock}>children</GridTwoColumn>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
