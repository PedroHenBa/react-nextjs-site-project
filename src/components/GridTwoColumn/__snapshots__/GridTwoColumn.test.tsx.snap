// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render Two Column Grid component 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4 c5"
        >
          Grid two column
        </h2>
        <div
          class="c6"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec quis enim magna.
    Nam sit amet ante vitae massa pellentesque volutpat. Vestibulum sagittis justo sit amet pretium inte
    rdum. Phasellus ornare.
        </div>
      </div>
      <div
        class="styles__ImageContainer-sc-iqf48w-2"
      >
        <img
          alt="Grid two column"
          class="styles__Image-sc-iqf48w-3"
          src="assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
